src/caffe/solver.cpp

```diff
@@ -775,9 +776,101 @@ void AdaGradSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
   }
 }
 
+
+template <typename Dtype>
+void AdamSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
```

Comment:
Use const vector<Blob<Dtype>*>& net_params = this->net_->learnable_params(); instead.